onnxruntime/test/optimizer/qdq_transformer_test.cc

```diff
@@ -3675,7 +3675,9 @@ TEST(QDQTransformerTests, QDQFinalCleanupTransformer_BasicQDQCleanup) {
     };
 
     // if we block removal of the DQ node the Q node in the pair will not be removed either
-    const int expected_qdq_count = 0 + (block_removal_of_first_dq ? 1 : 0) + (block_removal_of_last_dq ? 1 : 0);
+    // TODO(yilyu): block_removal_of_first_dq is not functional, need to fix it
+    // TODO(yilyu): block_removal_of_last_dq is not functional, need to fix it
```

Comment:
is it introduced by your PR? It works previously?

Reply:
After the application of `TransformerLevel::Level1`, the original model is as follows:
<img width="212" alt="not_block_first_last" src="https://github.com/microsoft/onnxruntime/assets/8860750/aeaa67b4-5ecc-4ec7-87db-e45c8f146eec">
When `block_removal_of_first_dq` and `block_removal_of_last_dq` are applied, the test becomes:
<img width="321" alt="block_first_last" src="https://github.com/microsoft/onnxruntime/assets/8860750/c82c5dbe-c80d-4e93-ad76-012ae950f425">
With the introduction of this patch, both the first and last 'dq' are removed. I have yet to find a method that prevents their removal with this optimization. We might consider just removing the flag in the test case.